components/core/tree/nz-tree-base.service.ts

```diff
@@ -493,14 +494,19 @@ export class NzTreeBaseService {
 
   conductSelectedKeys(keys: NzTreeNodeKey[], isMulti: boolean): void {
     this.selectedNodeList = [];
+    this.isAlreadySelected = false;
     const calc = (nodes: NzTreeNode[]): boolean => {
       return nodes.every(node => {
```

Comment:
refer to #5431 ，the reason is we didn't reset the existing selected nodes. https://github.com/NG-ZORRO/ng-zorro-antd/pull/5431/files#diff-816a63f3e324b0d84ca78723010b557aR505

Reply:
Your job is better than mine!